File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/player/IAudioPlayer.java
package ru.testsimpleapps.coloraudioplayer.managers.player;

public interface IAudioPlayer {

    /*
    * Play with result success.
    * */
    boolean play();

    /*
    * Pause current track.
    * */
    boolean pause();

    /*
    * Next with result, has next.
    * */
    boolean next();

    /*
    * Previous with result, has previous.
    * */
    boolean previous();

    /*
    * Seek to position.
    * */
    boolean seek(int position);

    /*
    * Stop current. Reset state.
    * */
    boolean stop();

    /*
    * Set new track
    * */
    void setTrackPath(final String path);

    /*
    * Release android media player and other resources.
    * */
    void release();
}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/player/IAudioPlayer.kt
package ru.testsimpleapps.coloraudioplayer.managers.player

interface IAudioPlayer {

    /*
    * Play with result success.
    * */
    fun play(): Boolean

    /*
    * Pause current track.
    * */
    fun pause(): Boolean

    /*
    * Next with result, has next.
    * */
    fun next(): Boolean

    /*
    * Previous with result, has image_previous.
    * */
    fun previous(): Boolean

    /*
    * Seek to position.
    * */
    fun seek(position: Int): Boolean

    /*
    * Stop current. Reset state.
    * */
    fun stop(): Boolean

    /*
    * Set new track
    * */
    fun setTrackPath(path: String)

    /*
    * Release android media player and other resources.
    * */
    fun release()
}
